include/odp/api/spec/timer.h

```diff
@@ -247,19 +247,20 @@ odp_timer_t odp_timer_alloc(odp_timer_pool_t timer_pool, odp_queue_t queue, cons
 /**
  * Free a timer
  *
- * Free (destroy) a timer, reclaiming associated resources.
- * The timeout event for an active timer will be returned.
- * The timeout event for an expired timer will not be returned. It is the
- * responsibility of the application to handle this timeout when it is received.
+ * Frees a previously allocated timer. The timer must be inactive when calling this function.
+ * In other words, the application must cancel an active single shot timer (odp_timer_cancel())
+ * successfully or wait it to expire before freeing it. Similarly for a period timer, the
+ * application must cancel it (odp_timer_periodic_cancel()) and receive the last event from
+ * the timer (odp_timer_periodic_ack()) before freeing it.
  *
- * A periodic timer must be cancelled successfully before freeing it.
+ * In normal conditions, the call should always succeed to free an inactive timer.
```

Comment:
Is this statement useful at all? Of course the caller expects this to succeed most of the time.

Reply:
It means that timer free should not return failure as "busy freeing timers, try again later". It should return failure only on bad handle, timer pool destroyed already, etc exceptional - where later retry won't help.

Reply:
Updated the wording in v3.